### ISSUE TITLE
Sagemaker Training Job Support

### ISSUE THREAD

**Issue by mathephysicist** [dmlc/gluon-nlp#1283]
In order to drive usability of the platform I propose that we make all scripts have sagemaker support.

1)Rename args, or allow the correspond args to be very easy to wrap around.
2)Reduce need for 'store_true' args.
3)Have an argument which allows you to control where artifacts are stored, or datasets are grabbed.

**Comment by sxjscience** [comment]
I think this is doable. Would you know if there is an easy way to open a notebook like Colab in SageMaker?

**Comment by szha** [comment]
@mathephysicist could you also provide some context on what the interface should look like? Any reference doc?

@sxjscience SageMaker training job is different from the notebook

**Comment by sxjscience** [comment]
@mathephysicist Yes, it will be very helpful if you can provide us with some details.

**Comment by mathephysicist** [comment]
Sorry for the delay, in order to make Amazon SageMaker training easy we would need the following:
•	Create a dataset-root/model-root location so we choose where we grab datasets from (rather than assuming they are in a certain location.
•	Create a flag to specify where we save model-artifacts and checkpoints
•	Ensure training works on SageMaker Training (sometimes there are problems with Horovod and downloading pre-trained models). 
•	Do one of the 2 following solutions:
1.	Modify the arguments so that SageMaker training can work more easily.
a.	Modify, input arguments so they align with the arguments SageMaker needs for training.
    parser.add_argument('--model-dir', type=str, default=os.environ['SM_MODEL_DIR'])
   	  parser.add_argument('--train', type=str, default=os.environ['SM_CHANNEL_TRAIN'])
   	  parser.add_argument('--test', type=str, default=os.environ['SM_CHANNEL_TEST'])
        
	And some sort of checkpoints-save flag where we can choose where to save checkpoints.
2.	Probably the better solution (for integration into more platforms) 
a.	Create a “SageMaker Train” script which wraps our scripts and provides the functionality necessary for 1) (primarily, I list of environ variables, that it goes ahead and grabs.)

Let me know if I need to add more clarity. I'd be more than willing to help write the blog once we have worked out those kinks.

**Comment by sxjscience** [comment]
@mathephysicist I'm currently investigating whether it's possible to use the docker + SageMaker custom training job (https://github.com/aws/sagemaker-training-toolkit#create-a-docker-image-and-train-a-model) to do the training. Will try to get a POC. We have recently added docker support, which has mxnet + horovod + gluonnlp + tensorboard + jupyterlab installed: https://github.com/dmlc/gluon-nlp/tree/master/tools/docker. We can revise the Dockerfile to compile a runtime docker that works for Sagemaker
